src/ngraph/runtime/dynamic/dynamic_backend.cpp

```diff
@@ -79,30 +81,76 @@ bool runtime::dynamic::DynamicExecutable::call(
     // (1) all shapes;
     // (2) all values of shape-relevant input tensors.
 
+    NGRAPH_CHECK(m_wrapped_function->get_parameters().size() == inputs.size());
+
     std::vector<std::shared_ptr<runtime::Tensor>> wrapped_inputs;
     std::vector<element::Type> arg_element_types;
     std::vector<PartialShape> arg_shapes;
 
-    for (auto& input : inputs)
+    std::shared_ptr<Function> clone;
     {
-        if (auto dynamic_tensor = std::dynamic_pointer_cast<runtime::dynamic::DynamicTensor>(input))
-        {
-            NGRAPH_CHECK(dynamic_tensor->has_storage());
-            arg_element_types.push_back(dynamic_tensor->get_wrapped_tensor()->get_element_type());
-            arg_shapes.push_back(dynamic_tensor->get_wrapped_tensor()->get_shape());
-            wrapped_inputs.push_back(dynamic_tensor->get_wrapped_tensor());
-        }
-        else
+        // We'll use AlignedBuffers to back the base pointers, storing them in this vector for RAII
+        // purposes. It seems we need to wrap them in unique_ptr rather than store them directly in
```

Comment:
Agreed, move should not be disabled. I will  fix.

Reply:
Adding a `TODO` to address once #2949 is merged.